packages/tokens/non-fungible/src/extensions/enumerable/storage.rs

```diff
@@ -0,0 +1,536 @@
+
+/// Returns the total amount of tokens stored by the contract.
+///
+/// # Arguments
+///
+/// * `e` - Access to the Soroban environment.
+pub fn total_supply(e: &Env) -> Balance {
+    e.storage().instance().get(&StorageKey::TotalSupply).unwrap_or(0)
+}
+
+/// Returns the `token_id` owned by `owner` at a given `index` in the
+/// owner's local list. Use along with
+/// [`crate::NonFungibleToken::balance()`] to enumerate all of `owner`'s
+/// tokens.
+///
+/// # Arguments
+///
+/// * `e` - Access to the Soroban environment.
+/// * `owner` - Account of the token's owner.
+/// * `index` - Index of the token in the owner's local list.
+///
+/// # Errors
+///
+/// * [`NonFungibleTokenError::TokenNotFoundInOwnerList`] - When the token ID is
+///   not found in the owner's enumeration.
+pub fn get_owner_token_id(e: &Env, owner: &Address, index: TokenId) -> TokenId {
+    let key = StorageKey::OwnerTokens(OwnerTokensKey { owner: owner.clone(), index });
+    let Some(token_id) = e.storage().persistent().get::<_, TokenId>(&key) else {
+        panic_with_error!(e, NonFungibleTokenError::TokenNotFoundInOwnerList);
+    };
+
+    token_id
+}
+
+/// Returns the `token_id` at a given `index` in the global token list.
+/// Use along with [`NonFungibleEnumerable::total_supply()`] to enumerate
+/// all the tokens in the contract.
+///
+/// # Arguments
+///
+/// * `e` - Access to the Soroban environment.
+/// * `index` - Index of the token in the owner's local list.
+///
+/// # Notes
+///
+/// **IMPORTANT**: This function is only intended for non-sequential
+/// `token_id`s. For sequential `token_id`s, no need to call a function,
+/// the `token_id` itself acts as the global index.
+///
+/// # Errors
+///
+/// * [`NonFungibleTokenError::TokenNotFoundInGlobalList`] - When the token ID
+///   is not found in the global enumeration.
+pub fn get_token_id(e: &Env, index: TokenId) -> TokenId {
+    let key = StorageKey::GlobalTokens(index);
+    let Some(token_id) = e.storage().persistent().get::<_, TokenId>(&key) else {
+        panic_with_error!(e, NonFungibleTokenError::TokenNotFoundInGlobalList);
+    };
+
+    token_id
+}
+
+// ################## CHANGE STATE ##################
+
+/// Creates a token with the next available `token_id` and assigns it to `to`.
+/// Returns the `token_id` for the newly minted token.
+///
+/// # Arguments
+///
+/// * `e` - Access to the Soroban environment.
+/// * `to` - The address receiving the new token.
+///
+/// # Errors
+///
+/// * refer to [`crate::mintable::mint`] errors.
+/// * refer to [`increment_total_supply`] errors.
+///
+/// # Events
+///
+/// * topics - `["mint", to: Address]`
+/// * data - `[token_id: TokenId]`
+///
+/// # Notes
+///
+/// This is a wrapper around [`crate::mintable::sequential_mint()`], that also
+/// handles the storage updates for:
+/// * total supply
+pub fn sequential_mint(e: &Env, to: &Address) -> TokenId {
+    let token_id = mintable_sequential_mint(e, to);
```

Comment:
I'd suggest using `mintable::sequential_mint` as a more explicit way instead of an alias.